### PR TITLE
feat(auth): Allow global config to be located in XDG directory

### PR DESCRIPTION
**Pull request by elramen**
If the global config file does not exist in the home directory, look in the following paths:
- Linux: $XDG_CONFIG_HOME/sentry/sentrycli.ini  and $HOME/.config/sentry/sentrycli.ini
- Mac: $HOME/Library/Application Support/sentry/sentrycli.ini
- Windows: {FOLDERID_RoamingAppData}

This enables users to keep their home dir clean, adhering to standards such as XDG on linux.

Fixes GH-1521

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -527,10 +527,13 @@ impl Config {
...
 fn find_global_config_file() -> Result<PathBuf> {
     dirs::home_dir()
-        .ok_or_else(|| format_err!("Could not find home dir"))
-        .map(|mut path| {
-            path.push(CONFIG_RC_FILE_NAME);
-            path
+        .map(|p| p.join(CONFIG_RC_FILE_NAME))
+        .filter(|p| p.exists())
+        .or(dirs::config_dir()
+            .map(|p| p.join("sentry/sentrycli.ini"))
```
Comment (szokeasaurusrex) [inline src/config.rs:533]: We should extract `"sentry/sentrycli.ini"` into a constant, like `CONFIG_RC_FILE_NAME`

File: src/config.rs [hunk 2]
```diff
@@ -527,10 +527,13 @@ impl Config {
...
+            .filter(|p| p.exists()))
+        .ok_or_else(|| {
+            format_err!("Could not find config file. Please run `sentry-cli login` and try again!")
```
Comment (szokeasaurusrex) [inline src/config.rs:536]:
```suggestion
            format_err!("Could not find config file.")
```

Why add the note about running `sentry-cli login` here? I am unsure whether this is always necessary – auth token is not needed for all CLI requests. Even if we should be adding this, this change should be made in a separate PR, since it is unrelated to what we want to do in this change.

**Review by sl0thentr0py — APPROVED** [review]
